### DISCUSSION TITLE
Connecting LVGL button or RGB events to Hue/Matter/Mqtt

### DISCUSSION THREAD

**Discussion by jpmorrison** [arendst/Tasmota#21023] [Q&A]
I have an LCD touch screen working. The default gpio switch simply turns the display on/off, and this works in MQTT and Hue emulation/Alexa.

I want to do more than turn the display on/off so I’d like to setup event handlers for states that can be controlled in the Hue emulation or MQTT. I can try matter pairing if that’s a better approach.

For example, how can I expose RGB settings? I can set the colours in Berry but I don’t know how I can expose this so Hue/MQTT//Matter can see and control it.

```
lv.start()

hres = lv.get_hor_res()       # should be 240
vres = lv.get_ver_res()       # should be 240

scr = lv.scr_act()            # default screen object

#scr.set_style_bg_color(lv.color(lv.COLOR_GRAY), lv.PART_MAIN | lv.STATE_DEFAULT)

logo = lv.img(scr)
logo.set_tasmota_logo()
logo.center()


home_btn = lv.btn(scr)
home_btn.set_size(hres,vres)

btn_style = lv.style()
btn_style.set_bg_color(lv.color(0x1fa3ec))
#btn_style.set_bg_img_src(logo)   # doesn't work, prints garbage
home_btn.add_style(btn_style, lv.PART_MAIN | lv.STATE_DEFAULT)

home_status = 1

def btn_clicked_cb(obj, event)
    var btn = "Unknown"
    if   obj == home_btn  btn = "Home"
    end
    if home_status == 1   
       home_status = 0
       logo.move_background()
    else
       home_status = 1
       logo.move_foreground()
    end
    tasmota.cmd("Power on")
    print(btn, "button pressed, event=",event)
end 
```

**Comment by s-hadinger** [comment]
I'm sorry but I don't undestand what RGB colors you want to control. Just a few possibilities:
- for Hue emulation, you can have a Berry-controlled virtual device: https://tasmota.github.io/docs/Berry/#hue_bridge-module
- for Matter, you can define a virtual device also Berry-controlled: https://tasmota.github.io/docs/Matter/#matter-virtual-devices

**Reply by jpmorrison** [reply]
Thanks. I found a bit more info on discord but this helps. In my example it could be for setting the LCD color in set_bg_color().
But I still don't see how Berry "knows" something has changed. 

With the touchscreen I set an event callback  `home_btn.add_event_cb(btn_clicked_cb, lv.EVENT_CLICKED , 0)` so every time the screen is touched I get an event and can do something. 

Is there something like that for Hue or Matter? I want to write a callback for hue/matter. 

For Hue would I do something like this?

```
import hue_bridge

def l1_cb()
   print("something changed? ", self.get())
end

# declare a virtual light
l1 = light_state(light_state.DIMMER)
hue_bridge.add_light(11, l1, "Synthetic Dimmer", "V1", "Tasmota Factory")
l1.signal_change = l1_cb

```

**Reply by s-hadinger** [reply]
For Hue, it seems it's missing the callback or rules, I will check. The alternative is to do some polling on the `light_state` object.

For Matter, you receive normal Matter event that you can handle with Berry rules.

I will try to craft an example
